src/noVoidZeroRule.ts

```diff
@@ -0,0 +1,61 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+import * as tsutils from 'tsutils';
+
+import { ExtendedMetadata } from './utils/ExtendedMetadata';
+
+const FAILURE_STRING: string = 'Replace void 0 with undefined';
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static metadata: ExtendedMetadata = {
+        ruleName: 'no-void-zero',
```

Comment:
> `ruleName: 'no-void-zero',

IMO it's better to move away from prefixing rule names with `no-` for a couple of reasons:

* It discourages rules from later being modified to support the opposite case. That's not so relevant to this rule, but does for others _(e.g. `parameter-reassignment`; what if we want prefer reassigning them over making unnecessary new variables?)_
* It messes with sorting of rules, since some have `no-` and others don't

Reply:
👍 Makes total sense

Reply:
Do you think just `void-zero` is sufficient or `prefer-undefined`?

Reply:
`void-zero`. names around preferring undefined could get confused with rules in the `null` vs. `undefined` area.